src/fem/fe_system.cc

```diff
@@ -0,0 +1,416 @@
+    {
+      number_of_single_dofs[i] += fe->n_object_dofs(i, DOF_SINGLE);
+      number_of_pairs[i] += fe->n_object_dofs(i, DOF_PAIR);
+      number_of_triples[i] += fe->n_object_dofs(i, DOF_TRIPLE);
+      number_of_sextuples[i] += fe->n_object_dofs(i, DOF_SEXTUPLE);
+    }
+
+    for (int i=0; i<fe->get_generalized_support_points().size(); i++)
+        generalized_support_points.push_back(fe->get_generalized_support_points()[i]);
+
+    if (fe->polynomial_order() > order) order = fe->polynomial_order();
+    
+    for (int i=0; i<fe->n_dofs(); ++i)
+      fe_dof_indices_.push_back(DofComponentData(fe_index, i, comp_offset));
+    
+    // add indices of the basis functions to the vectors for the particular node
+    // we assume that in the base FE, the nodal dofs are ordered as follows:
+    //   dof 0 for node 0, dof 1 for node 1, ... dof dim for node dim,
+    //   dof dim+1 for node 0, dof dim+2 for node 1, ...
+    for (unsigned int i=0; i<fe->n_object_dofs(0, DOF_SINGLE); i++)
+      node_basis[i % (dim+1)].push_back(basis_offset+i);
+    
+    unsigned int n_cell_dofs = fe->n_object_dofs(dim, DOF_SINGLE)
+                              +fe->n_object_dofs(dim, DOF_PAIR)
+                              +fe->n_object_dofs(dim, DOF_TRIPLE)
+                              +fe->n_object_dofs(dim, DOF_SEXTUPLE);
+    // in base FE, first come the nodal dofs, then cell dofs
+    for (unsigned int i=0; i<n_cell_dofs; i++)
+      cell_basis.push_back(basis_offset+fe->n_object_dofs(0,DOF_SINGLE)+i);
+    
+    fe_index++;
+    comp_offset += fe->n_components();
+    basis_offset += fe->n_dofs();
+  }
+  
+  // make the permutation of dofs: first nodal dofs, then cell dofs
+  for (unsigned int i=0; i<dim+1; i++)
+    dof_basis_.insert(dof_basis_.end(), node_basis[i].begin(), node_basis[i].end());
+  dof_basis_.insert(dof_basis_.end(), cell_basis.begin(), cell_basis.end());
+  
+//   if (this->is_primitive_)
+//   {
+//     double dof_index = 0;
+//     // add index of nonzero component for each dof in FESystem
+//     for (auto fe : fe_)
+//     {
+//       for (int i=0; i<fe->n_dofs(); ++i)
+//         this->component_indices_.push_back(fe_dof_indices_[dof_index++].fe_index);
+//     }
+//   } else {
+    double dof_index = 0;
+    comp_offset = 0;
+    // add footprint of nonzero components for each dof in FESystem
+    for (auto fe : fe_)
+    {
+      for (int i=0; i<fe->n_dofs(); ++i)
+      {
+        std::vector<bool> nonzeros(n_components_, false);
+        for (unsigned int c=0; c<fe->n_components(); c++)\
+          nonzeros[comp_offset+c] = fe->get_nonzero_components(fe_dof_indices_[dof_index++].basis_index)[c];
+        this->nonzero_components_.push_back(nonzeros);
+      }
+      comp_offset += fe->n_components();
+    }
+//   }
+
+  compute_node_matrix();
+}
+
+
+
+template<unsigned int dim, unsigned int spacedim>
+double FESystem<dim,spacedim>::basis_value(const unsigned int i, const arma::vec::fixed<dim> &p) const
```

Comment:
This is confusing: 
- base_value returns scalar, even if called for FeSystem, and it probably fails for vector valued Fe
- do we need this method at all?

Reply:
Postponed to #887

Reply:
This seems simple make it now.

Reply:
Done